File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 09/07/2026, 10:13:22
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int original = x;
        int reverse = 0;
        while (x != 0) {
            int digit = x % 10;
            reverse = reverse * 10 + digit;
            x /= 10;
        }
        return original == reverse;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 20/07/2026, 10:35:03
class Solution(object):
    def isPalindrome(self, x):
        # Negative numbers are not palindrome
        # Numbers ending with 0 (but not 0 itself) are not palindrome
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        
        reversed_half = 0
        
        while x > reversed_half:
            reversed_half = reversed_half * 10 + x % 10
            x //= 10
        
        # For even digits: x == reversed_half
        # For odd digits: x == reversed_half // 10
        return x == reversed_half or x == reversed_half // 10